shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -113,66 +113,73 @@ async fn test_hot_reload_with_old_instance_shutdown() {
-    // This reduces race conditions since we're furthest from drain boundaries.
-    tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
-
-    // Track connection failures over multiple drain cycles
-    // The gradual shutdown drains 10% of initial connections (rounded up to at least 1)
-    // For 13 connections: 10% = 1.3 -> rounds to 2 connections per cycle
-    // Each drain cycle waits 10 seconds between drains
+    // After the new shotover starts, gradual shutdown begins immediately.
+    // The implementation drains connections continuously in chunks:
+    // - Chunk duration: 200ms
+    // - For 10s shutdown: 50 chunks
+    // - For 13 connections: ceil(13/50) = 1 connection per chunk
     //
-    // Note: Low connection counts give non-ideal cycle counts due to rounding.
-    // For example, 13 connections requires 7 cycles to fully drain.
-    // With realistic connection counts (100+), a 10% drain rate consistently takes ~10 cycles.
+    // Timeline: Every 200ms, 1 connection is drained continuously
+    // We'll verify the drain behavior at specific time intervals
     let total_connections = connections.len();
-    let expected_drain_per_cycle =
-        std::cmp::max(1, (total_connections as f64 * 0.1).ceil() as usize);
+    let shutdown_duration = Duration::from_secs(10);
+    let chunk_duration = Duration::from_millis(200);
+    let num_chunks = shutdown_duration.div_duration_f64(chunk_duration) as usize;
+    let connections_per_chunk = std::cmp::max(1, total_connections.div_ceil(num_chunks));
 
     // Verify the drain rate calculation matches expected value for 13 connections
     assert_eq!(
-        expected_drain_per_cycle, 2,
-        "Expected 2 connections to drain per cycle (10% of 13), but got {}",
-        expected_drain_per_cycle
+        connections_per_chunk, 1,
+        "Expected 1 connection to drain per chunk (ceil(13/50)), but got {}",
+        connections_per_chunk
     );
 
-    // Calculate how many drain cycles we need to verify
-    // We'll verify enough cycles to drain at least 50% of connections
-    let num_cycles_to_verify = std::cmp::min(
-        5,
-        (total_connections as f64 / expected_drain_per_cycle as f64 / 2.0).ceil() as usize,
-    );
-
-    // Verify we're testing the expected number of cycles for 13 connections
-    assert_eq!(
-        num_cycles_to_verify, 4,
-        "Expected to verify 4 drain cycles (50% of 13 connections / 2 per cycle), but got {}",
-        num_cycles_to_verify
-    );
-
-    // Verify each drain cycle
-    for cycle in 1..=num_cycles_to_verify {
+    // We'll check the state after a few drain cycles
+    // Each cycle consists of multiple chunks, so we need to wait appropriately
+    // Let's verify after approximately 20%, 40%, 60%, and 80% of the shutdown period
+    let check_interval = chunk_duration * 10; // 10 chunks = ~2 seconds per check
+    let check_intervals = [
```

Comment:
do we need this array if all the intervals are the same?
Maybe just do `for i in 0..4 {` for the loop below?